pkgs/tools/misc/flowgger/default.nix

```diff
@@ -0,0 +1,46 @@
+{ lib
+, rustPlatform
+, fetchFromGitHub
+, pkg-config
+, openssl
+, capnproto
+}:
+
+rustPlatform.buildRustPackage rec {
+  pname = "flowgger";
+  version = "0.3.0";
+
+  src = fetchFromGitHub {
+    owner = "awslabs";
+    repo = pname;
+    rev = version;
+    hash = "sha256-xLNz3UW+s9lZA43Ir3j6B7cIdvNzfUY4dz3WMtJ6PN0=";
+  };
+
+  cargoLock = {
+    lockFile = ./Cargo.lock;
+  };
+
+  postPatch = ''
+    cp ${./Cargo.lock} Cargo.lock
+  '';
+
+  nativeBuildInputs = [
+    pkg-config
+    capnproto
+  ];
+
+  buildInputs = [ openssl ];
+
+  passthru.updateScript = ./update.sh;
+
+  # test failed
```

Comment:
Why did tests fail?

Reply:
This is the error message.

```
failures:

---- flowgger::encoder::ltsv_encoder::test_ltsv_full_encode_multiple_sd stdout ----
thread 'flowgger::encoder::ltsv_encoder::test_ltsv_full_encode_multiple_sd' panicked at 'assertion failed: `(left == right)`
  left: `"a:b\tc:123456\ta2:b2\tc2:123456\thost:testhostname\ttime:1691320524\tmessage:some test message\tfull_message:<23>Aug  6 11:15:24 testhostname appname[69]: 42 [someid a=\"b\" c=\"123456\"][someid2 a2=\"b2\" c2=\"123456\"] some test message\tlevel:7\tfacility:2\tappname:appname\tprocid:69\tmsgid:42"`,
 right: `"a:b\tc:123456\ta2:b2\tc2:123456\thost:testhostname\ttime:1659784524\tmessage:some test message\tfull_message:<23>Aug  6 11:15:24 testhostname appname[69]: 42 [someid a=\"b\" c=\"123456\"][someid2 a2=\"b2\" c2=\"123456\"] some test message\tlevel:7\tfacility:2\tappname:appname\tprocid:69\tmsgid:42"`', src/flowgger/encoder/ltsv_encoder.rs:196:5

---- flowgger::encoder::ltsv_encoder::test_ltsv_full_encode_no_sd stdout ----
thread 'flowgger::encoder::ltsv_encoder::test_ltsv_full_encode_no_sd' panicked at 'assertion failed: `(left == right)`
  left: `"host:testhostname\ttime:1691320524\tmessage:some test message\tfull_message:<23>Aug  6 11:15:24 testhostname appname[69]: 42 - some test message\tlevel:7\tfacility:2\tappname:appname\tprocid:69\tmsgid:42"`,
 right: `"host:testhostname\ttime:1659784524\tmessage:some test message\tfull_message:<23>Aug  6 11:15:24 testhostname appname[69]: 42 - some test message\tlevel:7\tfacility:2\tappname:appname\tprocid:69\tmsgid:42"`', src/flowgger/encoder/ltsv_encoder.rs:156:5


failures:
    flowgger::encoder::ltsv_encoder::test_ltsv_full_encode_multiple_sd
    flowgger::encoder::ltsv_encoder::test_ltsv_full_encode_no_sd

test result: FAILED(B. 87 passed; 2 failed; 0 ignored; 0 measured; 0 filtered out; finished in 0.30s
```

Reply:
I don't know what went wrong

Reply:
I'm not sure what caused this error, but how about using `checkFlags` to skip it?
https://github.com/NixOS/nixpkgs/blob/master/doc/languages-frameworks/rust.section.md#disabling-package-tests-disabling-package-tests

Reply:
Two tests were skipped.